lib/ansible/module_utils/basic.py

```diff
@@ -1321,7 +1321,9 @@ def set_mode_if_different(self, path, mode, changed, diff=None, expand=True):
                         if underlying_stat.st_mode != new_underlying_stat.st_mode:
                             os.chmod(b_path, stat.S_IMODE(underlying_stat.st_mode))
             except OSError as e:
-                if os.path.islink(b_path) and e.errno == errno.EPERM:  # Can't set mode on symbolic links
+                if os.path.islink(b_path) and e.errno in (errno.EACCES, errno.EPERM):
```

Comment:
Can we multiline this so that comments would go next to constants?
```suggestion
                if os.path.islink(b_path) and e.errno in (
                    errno.EACCES,  # can't access symlink in sticky directory (stat)
                    errno.EPERM,  # can't set mode on symbolic links (chmod)
                ):
```

Reply:
(it is recommended to put comments on the same line because this way it's less likely that some extra content appears between code and comments that way)